src/components/DatePickerRange.react.js

```diff
@@ -0,0 +1,298 @@
+import React, {Component, PropTypes} from 'react';
+import { DateRangePicker } from 'react-dates';
+import moment from 'moment';
+
+/**
+ * DatePickerRange is a tailor made component designed for selecting
+ * timespan across multiple days off of a calendar.
+ *
+ * The DatePicker integrates well with the Python datetime module with the
+ * startDate and endDate being returned in a string format suitable for
+ * creating datetime objects.
+ */
+export default class DatePickerRange extends Component {
+    constructor(props) {
+      super(props);
+      this.props.startDate = this.props.start_date;
+      this.props.endDate = this.props.end_date;
+      this.props.initialVisibleMonth = this.props.initial_visible_month;
+      this.props.minDateRange = this.props.min_date_range;
+      this.props.maxDateRange = this.props.max_date_range;
+      const momentProps = this.convertPropsToMoment(props);
+      this.state = {
+        focusedInput: props.autoFocusEndDate ? momentProps.startDate : momentProps.endDate,
+        startDate: momentProps.startDate,
+        endDate: momentProps.endDate,
+        initialVisibleMonth: momentProps.initialVisibleMonth,
+        minDateRange: momentProps.min,
+        maxDateRange: momentProps.max,
+        initialStartDate: this.props.start_date,
+        initialEndDate: this.props.end_date,
+        firstInitialVisibleMonth: this.props.initial_visible_month,
+        initialMinDateRange: this.props.min_date_range,
+        initialMaxDateRange: this.props.max_date_range
+      };
+
+    }
+
+    convertPropsToMoment(props) {
+      let startDate, endDate=null; let initialVisibleMonth = moment(props.startDate);
+      if(props.startDate != undefined) { startDate = moment(props.startDate); }
```

Comment:
I'm not sure if `undefined` is the right check here. Should this be a check against `null`? If it is `undefined`, then it should be `typeof props.startDate !== 'undefined'` i believe